### PR TITLE
feat: support aws provider 4.x

### PR DESCRIPTION
**Pull request by naseemkullah**
## Summary

Support 4.x of aws provider, see https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md

## How did you test this change?

vendored this module into our repo and adjusted the provider restraint for aws

## Issue

https://lacework.atlassian.net/jira/software/projects/ALLY/boards/58?selectedIssue=ALLY-915

### CODE REVIEW

File: versions.tf [hunk 1]
```diff
@@ -2,7 +2,7 @@ terraform {
   required_version = ">= 0.12.31"
 
   required_providers {
-    aws    = "~> 3.35"
+    aws    = ">= 3.35"
```
Comment (afiune) [inline versions.tf:5]: Could we add a more restricted constraint, we don't know if `5.0` will have breaking changes that will affect this module, what we have done in other modules is https://github.com/lacework/terraform-gcp-config/blob/main/versions.tf#L5
```suggestion
    aws    = ">= 3.35.0, < 5.0.0"
```

**Review by afiune — CHANGES_REQUESTED** [review]

**Comment by naseemkullah** [comment]
Constraint updated as per your suggestion @afiune

**Review by afiune — APPROVED** [review]
tenor-42629871

**Comment by naseemkullah** [comment]
thanks! any plan to cut a release soon?

**Pull request merged**